secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/domesticscheduledpayments/DomesticScheduledPaymentConsentsApiController.java

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.ob.uk.rs.server.api.obie.payment.v4_0_0.domesticscheduledpayments;
+
+import java.security.Principal;
+import java.util.Collections;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+import com.forgerock.sapi.gateway.ob.uk.common.datamodel.common.FRCharge;
+import com.forgerock.sapi.gateway.ob.uk.common.datamodel.converter.v4.payment.FRWriteDomesticScheduledConsentConverter;
+import com.forgerock.sapi.gateway.ob.uk.common.error.OBErrorResponseException;
+import com.forgerock.sapi.gateway.ob.uk.rs.obie.api.payment.v4_0_0.domesticscheduledpayments.DomesticScheduledPaymentConsentsApi;
+import com.forgerock.sapi.gateway.ob.uk.rs.server.api.obie.payment.factories.v4_0_0.OBWriteDomesticScheduledConsentResponse5Factory;
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.OBValidationService;
+import com.forgerock.sapi.gateway.rcs.consent.store.client.payment.domesticscheduled.DomesticScheduledPaymentConsentStoreClient;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.domesticscheduled.v3_1_10.CreateDomesticScheduledPaymentConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.domesticscheduled.v3_1_10.DomesticScheduledPaymentConsent;
+
+import jakarta.servlet.http.HttpServletRequest;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticScheduledConsent4;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticScheduledConsentResponse5;
+
+@Controller("DomesticScheduledPaymentConsentsApiV4.0.0")
+public class DomesticScheduledPaymentConsentsApiController implements DomesticScheduledPaymentConsentsApi {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final DomesticScheduledPaymentConsentStoreClient consentStoreApiClient;
+
+    private final OBValidationService<OBWriteDomesticScheduledConsent4> consentValidator;
+
+    private final OBWriteDomesticScheduledConsentResponse5Factory consentResponseFactory;
+
+    public DomesticScheduledPaymentConsentsApiController(
+            @Qualifier("v4.0.0RestDomesticScheduledPaymentConsentStoreClient") DomesticScheduledPaymentConsentStoreClient consentStoreApiClient,
+            OBValidationService<OBWriteDomesticScheduledConsent4> consentValidator,
+            OBWriteDomesticScheduledConsentResponse5Factory consentResponseFactory) {
+
+        this.consentStoreApiClient = consentStoreApiClient;
```

Comment:
null-checks for these three fields?

Reply:
Each one has null checks inside the OB objects (as previously mentioned), services, factories, etc.
Plus, the v4_0_0 controllers should follow the same pattern as the v3_1_10 controllers.